web/src/message_view.ts

```diff
@@ -208,7 +233,9 @@ function create_and_update_message_list(filter, id_info, opts) {
     // workflow we have which calls `message_view.show` after hash is updated.
     if (opts.change_hash) {
         update_hash_to_match_filter(filter, opts.trigger);
-        opts.show_more_topics = window.history.state?.show_more_topics ?? false;
+        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
```

Comment:
We currently do this in `browser_history.update_current_history_state_data` and I didn't see any other way to about it.

Reply:
Oh ick, no, the way is to parse it with Zod.

Reply:
Oh right right. So make a Zod object for `StateData` and parse it in both places. I can set that up

Reply:
I think this should be calling a function exported by the `browser_history` module; we shouldn't be randomly reaching into the internals of that data structure from here.

Reply:
Something like this?

```js
export function get_current_state_show_more_topics(): boolean | undefined {
    const current_state = state_data_schema.optional().parse(window.history.state);
    return current_state?.show_more_topics;
}
```

Reply:
Sure.